src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -130,6 +143,24 @@ public static LockFileTargetLibrary GetLibrary(this LockFileTarget lockFileTarge
             }
         }
 
+        private static void CollectDependencies(
```

Comment:
This method is almost a complete duplicate of the method above it.  Can you refactor to share code here?